lib/jsdom/living/nodes/HTMLTableCellElement-impl.js

```diff
@@ -34,19 +29,49 @@ class HTMLTableCellImpl extends HTMLElementImpl {
 
   get rowSpan() {
     const value = this.getAttribute("rowspan");
-    return value === null ? 1 : value;
+    if (value === null) {
+      return 1;
+    }
+    const parsed = parseInt(value);
+    if (isNaN(parsed)) {
+      return 1;
+    }
+    if (parsed < 0) {
+      return 0;
+    }
+    if (parsed > 65534) {
+      return 65534;
+    }
+    return parsed;
   }
 
   set rowSpan(V) {
     this.setAttribute("rowspan", String(V));
   }
 
-  _attrModified(name, value, oldValue) {
-    if (name === "headers" && this._headers) {
-      resetDOMTokenList(this._headers, value);
+  get cellIndex() {
+    const tr = closest(this, "tr", "http://www.w3.org/1999/xhtml");
+    if (tr === null) {
+      return -1;
+    }
+
+    return tr.cells.indexOf(this);
+  }
+
+  get scope() {
+    // TODO: This logic should only apply to <th>, not <td>, since only <th> defines the scope attribute and its states.
+    // But Firefox and Chrome alike seem to apply it to <td> as well.
+    // https://html.spec.whatwg.org/#attr-th-scope
```

Comment:
Correct me if I'm wrong about this of course, but IMO this should be true:

```js
// From the test I added
const th = document.createElement("th");
th.scope = "row";
console.assert(th.scope === "row");

const td = document.createElement("td");
td.scope = "row";
console.assert(td.scope === "");
```

In Firefox and Chrome only the first case is true. In the second case `td.scope` equates to `"row"` even though `td` elements should not support the `scope` attribute.

Reply:
What's the TODO here? The spec for HTMLTableCellElement's scope is that it reflects the scope content attribute. Chrome and Firefox seem to be working as intended.

I guess the spec could add a note that even though the scope content attribute is defined for th, it is reflected on tds anyway. But this implementation seems correct.

Reply:
> I guess the spec could add a note that even though the scope content attribute is defined for th, it is reflected on tds anyway.

That is exactly the bug. Even though the `scope` IDL attribute is defined for td, it defers to the "known values" for the specific element. Its known values are not defined for td (only for th), so the default known values of the empty set should be used for td. Yet browsers still apply the known values of `scope` for th to td.

Reply:
Well, known values are really for the attribute, not the element. At least that's the intent.

Reply:
> known values are really for the attribute, not the element

Indeed, but `td` element doesn't have a `scope` attribute...

----

Alright I'll change it. Do note though that this part of spec can be seen as fairly ambiguous.